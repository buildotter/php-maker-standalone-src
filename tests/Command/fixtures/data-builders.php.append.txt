<?php

declare(strict_types=1);

use Buildotter\Core\RandomMultiple;
use Buildotter\Tests\MakerStandalone\fixtures\expected\FooBuilder;

function random(): \Faker\Generator
{
    return \Faker\Factory::create();
}

function aFoo(): FooBuilder
{
    return FooBuilder::random();
}

/**
 * @return Buildotter\Tests\MakerStandalone\fixtures\Foo[]
 */
function someFoo(int|null $numberOfItems = null): array
{
    return RandomMultiple::from(FooBuilder::class, $numberOfItems);
}

/**
 * @return FooBuilder[]
 */
function someFooToBuild(int|null $numberOfItems = null): array
{
    return RandomMultiple::toBuildFrom(FooBuilder::class, $numberOfItems);
}
