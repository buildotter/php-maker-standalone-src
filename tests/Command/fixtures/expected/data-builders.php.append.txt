<?php

declare(strict_types=1);

use Buildotter\Core\Many;
use Buildotter\Tests\MakerStandalone\fixtures\expected\BarBuilder;
use Buildotter\Tests\MakerStandalone\fixtures\expected\FooBuilder;

function random(): Faker\Generator
{
    return Faker\Factory::create();
}

function aFoo(): FooBuilder
{
    return FooBuilder::new();
}

/**
 * @return Buildotter\Tests\MakerStandalone\fixtures\Foo[]
 */
function someFoos(int|null $numberOfItems = null): array
{
    return Many::from(FooBuilder::class, $numberOfItems);
}

/**
 * @return FooBuilder[]
 */
function someFoosToBuild(int|null $numberOfItems = null): array
{
    return Many::toBuildFrom(FooBuilder::class, $numberOfItems);
}

function aBar(): BarBuilder
{
    return BarBuilder::new();
}

/**
 * @return Buildotter\Tests\MakerStandalone\fixtures\Bar[]
 */
function someBars(int|null $numberOfItems = null): array
{
    return Many::from(BarBuilder::class, $numberOfItems);
}

/**
 * @return BarBuilder[]
 */
function someBarsToBuild(int|null $numberOfItems = null): array
{
    return Many::toBuildFrom(BarBuilder::class, $numberOfItems);
}
